sdk/typescript/test/e2e/read-transactions.test.ts

```diff
@@ -75,8 +75,8 @@ describe('Transaction Reading API', () => {
         }),
       ).rejects.toThrowError('The operation was aborted due to timeout');
 
-      // Expect it to poll the API at the provided interval
-      expect(spy.mock.calls).toHaveLength(6);
+      // Because JS event loop is somewhat unpredictable, we don't know exactly how long this will take, but we should have _at least_ 2 calls.
+      expect(spy.mock.calls.length).toBeGreaterThan(2);
```

Comment:
this is fine with me - I'm guessing a more "proper" solution would entail mocking a bunch of stuff or messing around with fake timers?

Reply:
Yeah fake timers would do this but I'm just being lazy for now